node/tests/node.rs

```diff
@@ -1,113 +1,85 @@
 #![cfg(not(target_arch = "wasm32"))]
 
-use std::env;
 use std::time::Duration;
 
 use celestia_node::{
     node::{Node, NodeConfig},
-    store::InMemoryStore,
-    test_utils::{test_node_config, test_node_config_with_keypair},
+    test_utils::{gen_filled_store, test_node_config, test_node_config_with_keypair},
 };
-use celestia_rpc::{prelude::*, Client};
-use libp2p::{identity, multiaddr::Protocol, Multiaddr, PeerId};
+use celestia_types::{consts::HASH_SIZE, hash::Hash};
+use libp2p::identity;
+use rand::Rng;
 use tokio::time::sleep;
 
-const WS_URL: &str = "ws://localhost:26658";
+use crate::utils::{fetch_bridge_info, new_connected_node};
 
-async fn fetch_bridge_info() -> (PeerId, Multiaddr) {
-    let _ = dotenvy::dotenv();
+mod utils;
 
-    let auth_token = env::var("CELESTIA_NODE_AUTH_TOKEN_ADMIN").unwrap();
-    let client = Client::new(WS_URL, Some(&auth_token)).await.unwrap();
-    let bridge_info = client.p2p_info().await.unwrap();
-
-    let mut ma = bridge_info
-        .addrs
-        .into_iter()
-        .find(|ma| ma.protocol_stack().any(|protocol| protocol == "tcp"))
-        .expect("Bridge doesn't listen on tcp");
-
-    if !ma.protocol_stack().any(|protocol| protocol == "p2p") {
-        ma.push(Protocol::P2p(bridge_info.id.into()))
-    }
+#[tokio::test]
+async fn connects_to_the_go_bridge_node() {
+    let node = new_connected_node().await;
 
-    (bridge_info.id.into(), ma)
+    let info = node.network_info().await.unwrap();
+    assert_eq!(info.num_peers(), 1);
 }
 
-async fn new_connected_node() -> Node<InMemoryStore> {
-    let (_, bridge_ma) = fetch_bridge_info().await;
-
+#[tokio::test]
+async fn header_store_access() {
+    let (store, _) = gen_filled_store(100);
     let node = Node::new(NodeConfig {
-        p2p_bootnodes: vec![bridge_ma],
+        store,
         ..test_node_config()
     })
     .await
     .unwrap();
 
-    node.p2p().wait_connected().await.unwrap();
-
-    // Wait until node reaches height 3
-    loop {
-        let head = node.p2p().get_head_header().await.unwrap();
-
-        if head.height().value() >= 3 {
-            break;
+    // check local head
+    let head = node.get_local_head_header().await.unwrap();
+    let expected_head = node.get_header_by_height(100).await.unwrap();
+    assert_eq!(head, expected_head);
+
+    // check getting existing headers
+    for _ in 0..100 {
+        let height = rand::thread_rng().gen_range(1..100);
```

Comment:
Why random?

Reply:
previously I was doing only 10 iterations there and picking random, now it's just a leftover

Reply:
removed